packages/calling-stateful-client/src/DeviceManager.test.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  AudioDeviceInfo,
+  AudioDeviceType,
+  DeviceAccess,
+  DeviceManager,
+  VideoDeviceInfo,
+  VideoDeviceType
+} from '@azure/communication-calling';
+import { CallContext } from './CallContext';
+import {
+  Mutable,
+  addMockEmitter,
+  createMockCallAgent,
+  createMockCallClient,
+  waitWithBreakCondition
+} from './TestUtils';
+import { createStatefulCallClientWithDeps, StatefulCallClient } from './StatefulCallClient';
+import { InternalCallContext } from './InternalCallContext';
+
+jest.mock('@azure/communication-calling');
+
+describe('device manager', () => {
+  test('should proxy getCameras and update cameras in state', async () => {
+    const client = createStatefulCallClientWithDeviceManager(
+      createMockDeviceManagerWithCameras([cameraWithName('camera')])
+    );
+
+    const cameras = await (await client.getDeviceManager()).getCameras();
+    expect(cameras.length).toBe(1);
+    expect(cameras[0].name).toBe('camera');
+
+    expect(client.getState().deviceManager.cameras.length).toBe(1);
+    expect(client.getState().deviceManager.cameras[0].name).toBe('camera');
+  });
+
+  test('should detect videoDevicesUpdated and update cameras in state using deviceManager', async () => {
+    const manager = createMockDeviceManagerWithCameras([cameraWithName('camera')]);
+    const client = createStatefulCallClientWithDeviceManager(manager);
+    // FIXME? Device manager is only created on request, so state is not updated unless device manager is created.
+    await client.getDeviceManager();
+
+    manager.emit('videoDevicesUpdated', {});
+    expect(await waitWithBreakCondition(() => client.getState().deviceManager.cameras.length === 1)).toBe(true);
+    expect(client.getState().deviceManager.cameras[0].name).toBe('camera');
```

Comment:
Thanks for taking the time to write these unit tests. Could you also check that `client.getState().deviceManager.cameras[0].name` to be 'camera'?

Reply:
That's line 47. Did I miss something?